src/pages/Rooms/_HeaderActions.js

```diff
@@ -1,24 +1,66 @@
 import React from 'react'
 import { connect } from 'react-redux'
 import history from '../../app/history'
-import { leaveRoomChannel } from '../../actions/rooms'
+import { archiveRoom, leaveRoomChannel, reactivateRoom } from '../../actions/rooms'
 import { deleteSubscription } from '../../actions/userSubscriptions'
 import notification from '../../helpers/notification'
 import { rootPath } from '../../helpers/paths'
-import { Menu, Dropdown, Button } from 'antd'
+import { Menu, Dropdown, Button, Modal } from 'antd'
 
-const RoomHeaderActions = ({ handleLeave, isSubscribed, isDirectMessage }) => {
+const RoomHeaderActions = ({ handleLeave, handleArchive, handleReactivate, isArchived, isSubscribed, isDirectMessage, room }) => {
   if (!isSubscribed || isDirectMessage) {
     return null
   }
 
+  const onArchiveConfirm = (event) => {
+    if (event) { event.preventDefault() }
+
+    Modal.confirm({
+      cancelText: 'Cancel',
+      content: 'This will disable all users from posting new messages and prevent new users from joining ' + room,
```

Comment:
Although it will say it right in the header, let's be redundant here:

> ~This~ Archiving a room will disable all users from posting new messages and prevent new users from joining